numpy/core/src/umath/ufunc_object.c

```diff
@@ -2343,6 +2344,16 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         }
     }
 
+    total_problem_size = NpyIter_GetIterSize(iter);
+    if (total_problem_size < 0) {
```

Comment:
The array isn't bigger, just the intermediate representation of the iterator before removing some axes. E.g. a call to `numpy.core.umath_tests.inner1d`, which has signature '(i),(i)->()', with two arrays of shapes `(a, 1, c)` and `(1, b, c)` will create an iterator with shape `(a, b, c, c)` before removing the last two dimensions. So with slightly more complicated gufuncs it is a meaningful possibility.